src/App.tsx

```diff
@@ -31,65 +31,113 @@ export function getReorderedGoods(
   goods: string[],
   { sortType, isReversed }: ReorderOptions,
 ) {
-  // To avoid the original array mutation
   const visibleGoods = [...goods];
 
-  // Sort and reverse goods if needed
-  // eslint-disable-next-line no-console
-  console.log(sortType, isReversed);
+  switch (sortType) {
+    case SortType.ALPHABET:
+      visibleGoods.sort((a, b) => a.localeCompare(b));
+      break;
+    case SortType.LENGTH:
+      visibleGoods.sort((a, b) => a.length - b.length);
+      break;
+    default:
+      // eslint-disable-next-line @typescript-eslint/no-unused-expressions
+      visibleGoods;
+  }
+
+  if (isReversed) {
+    visibleGoods.reverse();
+  }
 
   return visibleGoods;
 }
 
-// DON'T save goods to the state
-// type State = {
-//   isReversed: boolean,
-//   sortType: SortType,
-// };
-
-export const App: React.FC = () => {
-  return (
-    <div className="section content">
-      <div className="buttons">
-        <button
-          type="button"
-          className="button is-info is-light"
-        >
-          Sort alphabetically
-        </button>
-
-        <button
-          type="button"
-          className="button is-success is-light"
-        >
-          Sort by length
-        </button>
-
-        <button
-          type="button"
-          className="button is-warning is-light"
-        >
-          Reverse
-        </button>
-
-        <button
-          type="button"
-          className="button is-danger is-light"
-        >
-          Reset
-        </button>
-      </div>
+interface State {
+  isReversed: boolean,
+  sortType: SortType,
+}
+
+// eslint-disable-next-line react/prefer-stateless-function
+export class App extends React.Component<{}, State> {
+  state = {
+    isReversed: false,
+    sortType: SortType.NONE,
+  };
+
+  hadleClickAlphabetical = () => {
+    this.setState({ sortType: SortType.ALPHABET });
+  };
+
+  hadleClickLength = () => {
+    this.setState({ sortType: SortType.LENGTH });
+  };
+
+  hadleClickReverse = () => {
+    const { isReversed } = this.state;
+
+    if (isReversed) {
+      this.setState({ isReversed: false });
+    } else {
+      this.setState({ isReversed: true });
+    }
```

Comment:
you can always change the isReversed here to the opposite value with `!` and do it with 1 setState